src/core/cgroup.c

```diff
@@ -1725,16 +1699,14 @@ static void cgroup_xattr_apply(Unit *u) {
 static bool unit_has_mask_realized(
                 Unit *u,
                 CGroupMask target_mask,
-                CGroupMask enable_mask,
-                bool needs_bpf) {
+                CGroupMask enable_mask) {
 
         assert(u);
 
         return u->cgroup_realized &&
                 u->cgroup_realized_mask == target_mask &&
                 u->cgroup_enabled_mask == enable_mask &&
-                ((needs_bpf && u->cgroup_bpf_state == UNIT_CGROUP_BPF_ON) ||
-                 (!needs_bpf && u->cgroup_bpf_state == UNIT_CGROUP_BPF_OFF));
+                !u->cgroup_invalidated_mask;
```

Comment:
hmm, according to our CODING_STYLE we prefer not to rely on C's downgrade-to-bool feature when checking variables that are not booleans (or boolean-like). i.e. given that this is a bitmask please replace `!u->cgroup_invalidated_mask` → `u->cgroup_invalidated_mask != 0`...